core/service/src/vault/storage/mod.rs

```diff
@@ -72,6 +107,32 @@ pub trait Storage: StorageReader {
     async fn delete_data(&mut self, data_id: &RequestId, data_type: &str) -> anyhow::Result<()>;
 }
 
+/// Extended storage trait for epoch-aware data storage and deletion.
+///
+/// This trait combines [`StorageReaderExt`] and [`Storage`] with additional methods
+/// for storing and deleting data organized by epoch IDs. Use this trait when you need
+/// to manage private key material that may belong to a specific epoch.
+#[enum_dispatch]
+#[trait_variant::make(Send)]
+pub trait StorageExt: StorageReaderExt + Storage {
```

Comment:
I am not sure but I wonder if it might be more concise to instead have a single storage type with `epoch_id` as optional?

Reply:
indeed there are a bit too many traits in here.

I took a first step in simplifying them in this PR https://github.com/zama-ai/kms/pull/342 

it merged the bytes trait with the storage ones.

but the epoch ones are still independent from the normal ones. for now the public storage will never use epoch, but if it ever does we can merge the two traits.